### ISSUE TITLE
Aliasing issue resizing JPG

### ISSUE THREAD

**Issue by Kiruchi** [lovell/sharp#807]
First, thanks for the awesome work you did with Sharp ! 👍 

I have a question (I don't think it is a bug) regarding the resize function.
My code looks like this :
```javascript
const outputImage = sharp(imagePath);
outputImage
        .resize(Number(width), Number(height))
        .max();
```

When I run it on my Windows machine, the resized image is perfectly fine, without aliasing.

When I run it on a Debian 8 machine, it generates the following image : 
Source image : 
![test_source](https://cloud.githubusercontent.com/assets/3627263/26049389/172b6332-395b-11e7-9e10-a107c7938fb6.jpg)

Output with size 350x350 (the "bug" happens) : 
![test_350x350](https://cloud.githubusercontent.com/assets/3627263/26049403/288271de-395b-11e7-9f12-32f2f627aba4.jpg)

Output with size 351x351 (perfectly fine) : 
![test_351x351](https://cloud.githubusercontent.com/assets/3627263/26049408/2b3566fc-395b-11e7-9d2a-3a8664187950.jpg)

It seems to have this behaviour only with some "round" numbers and only on my Debian server, not on Windows.

Any help would be much appreciated.
Thanks !

**Comment by lovell** [comment]
Hello, this might be related to #781 - are you able to try using the latest code on the [ridge](https://github.com/lovell/sharp/tree/ridge) branch?
```sh
npm install lovell/sharp#ridge
```

**Comment by Kiruchi** [comment]
Hello lovell and thank you for your answer.
I did try with the branch you mentionned (If it might help someone else, I had to use libvips v. 8.5.4) for it to work).
I still face the same issue though.

**Comment by lovell** [comment]
Are you able to provide the original image as a file via e.g. Dropbox, Google Drive etc. as Github has processed your uploaded images.

**Comment by Kiruchi** [comment]
Here it is : https://drive.google.com/open?id=0B8lWt9uI1FA4Q3N6SHJwTVkyQ1U

**Comment by lovell** [comment]
Thank you.

Are you able to reproduce this when vips is not installed globally, so sharp will use its own pre-built version? (You'll see a `node-modules/sharp/vendor` directory appear at `npm install` time. This should work for the ridge branch too as the v8.5.4 binaries are available.)

**Comment by Kiruchi** [comment]
I can't reproduce on the Debian machine I am working on. It is a 32 bits Debian 8, I saw on the installation page that Sharp can only be installed on 32 bits Linux systems when Libvips is installed globally. Tell me if I'm wrong. (I think I tried without the global Libvips installation and it bugged)
Currently I have Libvips 8.5.4 globally installed and Sharp on Ridge branch.

**Comment by lovell** [comment]
My best guess would be rounding errors due to the use of the x87 FPU in 32-bit CPUs.

See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=323#c109 for some possible compiler flags to workaround this.

**Comment by lovell** [comment]
Enabling [simd](http://sharp.dimens.io/en/stable/api-utility/#simd) might help too as that will keep the floating-point reduction away from the x87 FPU.

**Comment by Kiruchi** [comment]
Thanks a lot for these tips. You were almost right and pointed me in the right direction. I had to follow libvips docs to "build with the GCC auto-vectorizer and diagnostics (or just -O3)" (from here : https://github.com/jcupitt/libvips) : 
```
$ FLAGS="-O2 -msse4.2 -ffast-math"
$ FLAGS="$FLAGS -ftree-vectorize -fdump-tree-vect-details"
$ CFLAGS="$FLAGS" CXXFLAGS="$FLAGS" \
	./configure --prefix=/home/john/vips 
```
It worked with these params.
For information I tried enabling "simd" first but it didn't work.

Again, thanks a lot for the time you give to the community, you are awesome and Sharp is amazing :)
